src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,1411 @@
+          // };
+          selectedSource = {
+            created_at: new Date().toISOString().replace('Z', '+00:00'), // technically not the exact time it shows up in the database
+            end_time: sourceInsert.end_time,
+            pkey: {
+              derivation_group_name: derivationGroupInsert.name,
+              key: sourceInsert.key,
+            },
+            source_type_name: sourceTypeInsert.name,
+            start_time: sourceInsert.start_time,
+            valid_at: sourceInsert.valid_at,
+          };
+          gridRowSizes = gridRowSizesBottomPanel;
+        }
+      }
+    } else {
+      showFailureToast('Upload failed.');
+    }
+    // Reset the form behind the source
+    parsed = undefined;
+    keyField.reset('');
+    sourceTypeField.reset('');
+    startTimeDoyField.reset('');
+    endTimeDoyField.reset('');
+    validAtDoyField.reset('');
+    derivationGroupField.reset('');
+  }
+
+  async function selectSource(detail: ExternalSourceSlim) {
+    selectedSource = detail;
+    gridRowSizes = gridRowSizesBottomPanel;
+    deselectEvent();
+    eventTooltip.reset();
+  }
+
+  function deselectSource() {
+    deselectEvent();
+    eventTooltip.reset();
+    gridRowSizes = gridRowSizesNoBottomPanel;
+    selectedSource = null;
+  }
+
+  function deselectEvent() {
+    selectedEvent = null;
+    selectedRowId = null;
+  }
+
+  function onCanvasMouseDown(e: CustomEvent<MouseDown>) {
+    if (!mouseDownAfterTable) {
+      const { externalEvents } = e.detail;
+      // selectedEvent is our source of an ExternalEvent as well as the ExternalEventId used by this instance
+      //    of the LayerExternalSources (as opposed to using a store, like the timeline one does, which is
+      //    unnecessary as everything we need is in on single component or can be passed down via parameters to
+      //    children).
+      selectedEvent = externalEvents?.length ? externalEvents[0] : null;
+      selectedRowId = null;
+    } else {
+      mouseDownAfterTable = false;
+    }
+  }
+
+  function onCanvasMouseOver(e: CustomEvent<MouseOver>) {
+    mouseOver = e.detail; // Allows tooltip to access object
+  }
+
+  function toggleItem(value: ExternalSourceType) {
+    if (!selectedFilters.find(f => f.name === value.name)) {
+      selectedFilters = selectedFilters.concat(value);
+    } else {
+      selectedFilters = selectedFilters.filter(filter => filter.name !== value.name);
+    }
+  }
+
+  function unselectFilteredValues() {
+    selectedFilters = [];
+  }
+
+  function selectFilteredValues() {
+    selectedFilters = [...new Set([...selectedFilters, ...filteredValues])];
+  }
+
+  function onSelectionChanged() {
+    selectedEvent = selectedEvents.find(event => getRowIdExternalEvent(event.pkey) === selectedRowId) ?? null;
+  }
+
+  function onManageGroupsAndTypes() {
+    effects.manageGroupsAndTypes(user);
+  }
+</script>
+
+<CssGrid columns={uiColumnSize}>
+  <Panel borderRight padBody={true}>
```

Comment:
Could we skip padding on the body of the panel (there's a prop for this) and use a fieldset for every child? Alternatively you could mirror what the Models page form does.

Reply:
I actually don't remember why we had `padBody={true}` on here - removing it doesn't seem to effect the styling. We can mirror the way Models implements a similar set of forms though, if it would be better to keep implementations consistent!

Reply:
Believe it defaults to `true` so setting it to `false` will remove the padding which is now probably what is desired in order to match the other similar forms.